### PR TITLE
New literal DateTime "earlier"/"before" = 1970-01-02

### PR DESCRIPTION
**Pull request by quazgar**
For #35 and https://github.com/GothenburgBitFactory/timewarrior/issues/407

### CODE REVIEW

File: test/datetime.t.cpp [hunk 1]
```diff
@@ -973,6 +973,11 @@ int main (int, char**)
     t.is (later.day (),    30, "later -> d = 30");
     t.is (later.year (), 9999, "later -> y = 9999");
 
+    Datetime earlier ("earlier");
+    t.is (earlier.month (),   1, "earlier -> m = 1");
+    t.is (earlier.day (),     1, "earlier -> d = 1");
+    t.is (earlier.year (), 1970, "earlier -> y = 1970");
+
```
Comment (lauft) [inline test/datetime.t.cpp:980]: As far as I see it, a test for "before" is missing. Something like
```
Datetime before ("before");
t.is (before.month (),   1, "before -> m = 1");
t.is (before.day (),     1, "before -> d = 1");
t.is (before.year (), 1970, "before -> y = 1970");
```
Yes, "before" is mapped to "earlier" in `Datetime.cpp:1626`, but without this, you only verify "before" *is* mapped,  not whether it is mapped correclty.

**Review by lauft — COMMENTED** [review]
I still have to make up my mind about this addition. Maybe @tbabej has some insights whether this is useful.

**Comment by tbabej** [comment]
I made some comments on https://github.com/GothenburgBitFactory/timewarrior/issues/407

File: test/datetime.t.cpp (hunk 1 above)
Comment (quazgar) [reply test/datetime.t.cpp]: Done, I am testing for `before == earlier`, and also added the same test for `someday == later`.

**Comment by lauft** [comment]
See https://github.com/GothenburgBitFactory/timewarrior/pull/408#issuecomment-859332788

**Pull request closed without merging**